app/queries/MeasurementQuery.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package queries
+
+import play.api.libs.json.JsPath
+
+case object MeasurementQuery extends Gettable[Map[String, String]] with Settable[Map[String, String]] {
```

Comment:
Can we do it the way the others do it

case class MeasurementQuery(recordId) extends Gettable[String] with Settable[String] {

override def path: JsPath = JsPath \ recordId \ "measurementUnit"
}

That makes the code where you have to get the value a lot simpler and it makes cleaning up the code a lot easier?

Reply:
done it